src/models/v3/client/profile.rs

```diff
@@ -0,0 +1,146 @@
+use std::path::PathBuf;
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    database,
+    models::ids::{Base62Id, UserId, VersionId},
+};
+
+/// The ID of a specific profile, encoded as base62 for usage in the API
+#[derive(Copy, Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(from = "Base62Id")]
+#[serde(into = "Base62Id")]
+pub struct ClientProfileId(pub u64);
+
+/// The ID of a specific profile link, encoded as base62 for usage in the API
+#[derive(Copy, Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(from = "Base62Id")]
+#[serde(into = "Base62Id")]
+pub struct ClientProfileLinkId(pub u64);
+
+/// A project returned from the API
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ClientProfile {
+    /// The ID of the profile, encoded as a base62 string.
+    pub id: ClientProfileId,
+
+    /// The person that has ownership of this profile.
+    pub owner_id: UserId,
+    /// The title or name of the project.
+    pub name: String,
+    /// The date at which the project was first created.
+    pub created: DateTime<Utc>,
+    /// The date at which the project was last updated (versions/override were added/removed)
+    pub updated: DateTime<Utc>,
+    /// The icon of the project.
+    pub icon_url: Option<String>,
+
+    // Users that are associated with this profile
+    // Hidden if the user is not the owner
+    pub users: Option<Vec<UserId>>,
+
+    /// The loader
+    pub loader: String,
+
+    /// Game-specific information
+    #[serde(flatten)]
+    pub game: ClientProfileMetadata,
+
+    /// Modrinth-associated versions
+    pub versions: Vec<VersionId>,
+    /// Overrides for this profile- only install paths are given,
+    /// hashes are looked up in the CDN by the client
+    pub override_install_paths: Vec<PathBuf>,
```

Comment:
What does this mean? IMO we shouldn't have the versions or any install info be returned here. It should be a separate route that is only called when the pack is being installed. And it can only be accessed once a user has accepted the invite.

Reply:
Im nott sure I agree- I think it could be useful to be able to see the stuff youre going to install on your computer before accepting, or what's 'in/on' the shared modpack.

All this displays is what the versions are and the 'names' of the overrides.